test/factories/pets.rb

```diff
@@ -14,12 +14,12 @@
     weight_unit { "lb" }
 
     trait :adoption_pending do
-      adopter_applications { build_list(:adopter_application, 3, :adoption_pending) }
+      adopter_applications { build_list(:adopter_application, 3, :adoption_pending, pet: instance) }
     end
 
     trait :adopted do
-      match { association :match }
-      adopter_applications { build_list(:adopter_application, 1, :successful_applicant) }
+      match { association :match, pet: instance }
+      adopter_applications { build_list(:adopter_application, 1, :successful_applicant, pet: instance) }
```

Comment:
This was an especially annoying bug. I realized that `FactoryBot.create(:pet, :adopted)` was instantiating 2 pets, not 1. You can try this in console in `main`:
```ruby
Pet.count # 10
FactoryBot.create(:pet, :adopted)
Pet.count #12
```
A second pet is instantiated within the association since the instance is not passed to it.